Coordinate/Coordinate.cs

```diff
@@ -17,5 +17,38 @@ public Coordinate(decimal latitude, decimal longitude)
             this.Latitude = latitude;
             this.Longitude = longitude;
         }
+
+        public override bool Equals(object obj)
+        {
+            return (obj as Coordinate? != null)
+                ? this.Equals((Coordinate)obj)
+                : false;
+        }
+
+        public bool Equals(Coordinate other)
+        {
+            return this.Latitude == other.Latitude
+                && this.Longitude == other.Longitude;
+        }
+
+        public override int GetHashCode()
+        {
+            return this.Latitude.GetHashCode() ^ this.Longitude.GetHashCode();
+        }
+
+        public static bool operator ==(Coordinate left, Coordinate right)
+        {
+            return left.Equals(right);
+        }
+
+        public static bool operator !=(Coordinate left, Coordinate right)
+        {
+            return !left.Equals(right);
+        }
+
+        public override string ToString()
+        {
+            return string.Format("{0}N, {1}E", Latitude, Longitude);

```

Comment:
Depends on if you want to deal with signs or not. The Google Maps API accepts North and East, using negative numbers for South and West, so I adopted that convention.